libqtile/state.py

```diff
@@ -45,6 +53,39 @@ def apply(self, qtile):
             Rearrange the windows in the specified Qtile object according to
             this QtileState.
         """
+        try:
+            for group in qtile.groups:
+                x = []
+                for i in self.focusHistory[group.name]:
+                    x.append(qtile.windowMap[i.wid])
+                group.focusHistory = x
+                for layout in group.layouts:
+                    d = self.layoutMap[group.name][layout.name]
+                    d.group = layout.group
+                    members = dir(d)
+                    for member in members:
```

Comment:
I'll admit, I haven't worked with the state code, but what is it you're trying to pull out here? Quickly searching, I don't see where a Window would end up in a Layout'- dir().